userspace/libsinsp/threadinfo.cpp

```diff
@@ -774,27 +774,15 @@ void sinsp_threadinfo::traverse_parent_state(std::function<bool (sinsp_threadinf
 	// state, at different rates. If they ever equal each other
 	// before slow is NULL there's a loop.
 
-	sinsp_threadinfo *slow=this, *fast=this;
+	sinsp_threadinfo *slow=this->get_parent_thread(), *fast=slow;
```

Comment:
It was never calling visitor on the thread itself. The prior version advanced slow before calling the visitor, so this behavior should be unchanged.